src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeature.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public class ViewsFeature
+    {
+        public IDictionary<string, Type> Views { get; } =
+            new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
```

Comment:
We use case insensitive lookups in other view lookup places (for instance the `CompilerCache`). Sadly the case sensitivity here wouldn't make a big difference.